### DISCUSSION TITLE
Hyundai base API

### DISCUSSION THREAD

**Discussion by Jesus-M** [Hyundai-Kia-Connect/hyundai_kia_connect_api#426] [General]
Hi!

I am trying to figure out how the base API works following what this wrapper does, but not too much progress yet, so I was wondering if there is any documentation at all of the base API itself that I could follow.

Thanks in advance.

**Comment by cdnninja** [comment]
So to confirm you want to know how the raw api works?   

What region? They are all different.

**Comment by Jesus-M** [comment]
Yes, kind of.

I am not that good with Python (being generous to myself) and it's hard to me to follow how a Python script goes. I am much better with other scripting languages, usually enough for my small projects (bash and perl, for example).

For what I am intending to do, I don't need a full integration of the car, just few parameters, and I am not sure if I will be over killing the system with too many unnecessary requests. That's why I am trying to understand the real requests to be able to minimize the number to the bare minimum I need, so I might be able to increase the frequency.

The final goal is to be able to integrate with my PV Solar panel installation, so I can make more efficient the charging of the car.

My region is Europe, Ireland more specifically. 

Thanks!

**Comment by cdnninja** [comment]
Most data comes from a single query.  Cached being ideal vs forced as it doesn't wake the car.   

For setting items it's one call per action.   

Honestly the login process is complicated.   Replicating it in another language will be time consuming.  Check out the eu file in this repo.   That is the entire call setup.  Many of the methods being login related for stamps.

**Comment by Jesus-M** [comment]
> Check out the eu file in this repo. That is the entire call setup. Many of the methods being login related for stamps.

That's what I found difficult, following the logic of the Python code on the file. But sure, I will have a look on that.

Thanks
